### DISCUSSION TITLE
Cocoapod support

### DISCUSSION THREAD

**Discussion by arshiacont** [orchetect/swift-midi#194] [Ideas]
First of all, thanks for sharing such a great package! Greatly reduces the pain of dealing with CoreMIDI!!!

While Swift Package Manager is the way to go these days, for some of us are still dealing with mixed C++/ObjC/ObjC++/Swift programing and SPM is [still] struggling with such libraries. Since this library is Swift Only supporting a Cocoapod version should be easy and would allow many of us to integrate with other pods.

Any thoughts of adding a podspec and allowing distribution on Cocoapods?

**Comment by orchetect** [comment]
Sure, if someone wanted to set it up and PR. I've had Cocoapods support for packages in the past and it's kind of a hassle especially with dependencies, and there isn't very much demand for it generally.

**Reply by arshiacont** [reply]
I can give it a try.. I guess you are referring to TimecodeKit and XCTestUtils dependencies?! In that case they both of them should support Cocoapods! Any other dependencies to look at?

**Reply by orchetect** [reply]
It's just TimecodeKit.

**Reply by arshiacont** [reply]
Just added PRs for both TimecodeKit and MIDIKit. And they build fine!
MIDIKit move to Cocoapods was a hassle! :) If there is no demand for the pod, we might as well avoid running `pod spec lint` (which appears to be full of bugs!) and keep the podspec in the repo.
If you believe this is messing with your code, you can also just reject the PR and I'll continue using the Fork until I move everything on my side to SPM (which is another hassle given the messy integration of ObjC/C++). I just wanted to keep up to the promise here!
Great job BTW.

**Reply by orchetect** [reply]
Thanks for helping out - like I say, I've managed podspecs for packages in the past (ones that had no dependencies) and it was working ok for a while. But a year or two ago the tooling had issues and nothing was passing linting and I kind of gave up at that point.

SPM is infinitely better than Cocoapods and everyone should be using it where possible (and you can probably see why even just through this process). But I understand for legacy codebases or compatibility some are forced to use Cocoapods.

Let's get TimecodeKit working first and then I can merge that PR main there. Then we can look at MIDIKit.
